src/backend/functions/projects-edit.ts

```diff
@@ -513,9 +513,11 @@ const handler = middy(editProject)
 export { handler };
 
 // Given a user's id, this method returns the user's full name
-const getUserFullName = async (userId: any): Promise<string> => {
+const getUserFullName = async (userId: number | null): Promise<string | null> => {
+  if (userId === null) {
+    return null; 
+  }
   const user = await prisma.user.findUnique( { where: { userId }});
   if (!user) throw new Error('user not found'); 
-  const userName = user ? `${user.firstName} ${user.lastName}` : '—';
-  return userName;  
+  return user ? `${user.firstName} ${user.lastName}` : '—';
 }
```

Comment:
<img width="960" alt="#555_2nd_proposed_solution_2" src="https://user-images.githubusercontent.com/91857946/176592834-06e87843-d954-45c1-8fc5-b29ef1b9ae52.png">
<img width="960" alt="#555_2nd_proposed_solution" src="https://user-images.githubusercontent.com/91857946/176592835-4f1fd4a1-f536-45d0-839f-e52132fe6d88.png">

*** I also had 2nd solution which I believe it should behave the same but I shrank the code so the code looks more simple by using nullish coalescing operator (??) at lines 189, 190, 203, and 204 in the screenshot. (also applied it in project-edit.ts file) I'm not sure if either the commit version or screenshot version is a good implementation but for now, I thought of these two solutions. Let me know your thoughts on both of these two versions.

Reply:
I think it makes most sense to not utilize -1 to represent something special. I also don't think we should return the em-dash when a valid userId is not supplied, because I think that impacts the behavior of the other functions

Reply:
I see. Right, forgot that we decided to throw an error in getUserFullName if user is not found from prisma. So Isn't that mean we can just return user's full name after we check the user exists from prisma? (Also did you had chance to review the committed version? Wondered if that version looks okay)